File: examples/android/src/main/java/com/dropbox/core/examples/android/UserActivity.java
package com.dropbox.core.examples.android;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.widget.Toolbar;

import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.dropbox.core.examples.android.internal.OpenWithActivity;
import com.dropbox.core.v2.users.FullAccount;

import java.util.Arrays;


/**
 * Activity that shows information about the currently logged in user
 */
public class UserActivity extends DropboxActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_user);

        String apiKey = getString(R.string.app_key);

        if (apiKey.equals("YOUR_API_KEY_HERE")) {
            new AlertDialog.Builder(this)
                    .setTitle("API KEY Required")
                    .setMessage("You must specify an API KEY in local.properties.\n" + "You can get your own API key at https://developers.dropbox.com/.")
                    .create()
                    .show();
        }

        Toolbar toolbar = (Toolbar) findViewById(R.id.app_bar);
        setSupportActionBar(toolbar);

        Button loginButton = (Button) findViewById(R.id.login_button);

        loginButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                DropboxActivity.startOAuth2Authentication(UserActivity.this, apiKey, Arrays.asList("account_info.read", "files.content.write"));
            }
        });

        Button filesButton = (Button) findViewById(R.id.files_button);
        filesButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(FilesActivity.getIntent(UserActivity.this, ""));
            }
        });

        Button openWithButton = (Button) findViewById(R.id.open_with);
        openWithButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent openWithIntent = new Intent(UserActivity.this, OpenWithActivity.class);
                startActivity(openWithIntent);
            }
        });
    }

    @Override
    protected void onResume() {
        super.onResume();

        if (hasToken()) {
            findViewById(R.id.login_button).setVisibility(View.GONE);
            findViewById(R.id.email_text).setVisibility(View.VISIBLE);
            findViewById(R.id.name_text).setVisibility(View.VISIBLE);
            findViewById(R.id.type_text).setVisibility(View.VISIBLE);
            findViewById(R.id.files_button).setEnabled(true);
            findViewById(R.id.open_with).setEnabled(true);
        } else {
            findViewById(R.id.login_button).setVisibility(View.VISIBLE);
            findViewById(R.id.email_text).setVisibility(View.GONE);
            findViewById(R.id.name_text).setVisibility(View.GONE);
            findViewById(R.id.type_text).setVisibility(View.GONE);
            findViewById(R.id.files_button).setEnabled(false);
            findViewById(R.id.open_with).setEnabled(false);
        }
    }

    @Override
    protected void loadData() {
        new GetCurrentAccountTask(DropboxClientFactory.getClient(), new GetCurrentAccountTask.Callback() {
            @Override
            public void onComplete(FullAccount result) {
                ((TextView) findViewById(R.id.email_text)).setText(result.getEmail());
                ((TextView) findViewById(R.id.name_text)).setText(result.getName().getDisplayName());
                ((TextView) findViewById(R.id.type_text)).setText(result.getAccountType().name());
            }

            @Override
            public void onError(Exception e) {
                Log.e(getClass().getName(), "Failed to get account details.", e);
            }
        }).execute();
    }

}

File: examples/android/src/main/java/com/dropbox/core/examples/android/UserActivity.kt
package com.dropbox.core.examples.android

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.widget.Toolbar
import com.dropbox.core.examples.android.internal.DropboxCredentialUtil
import com.dropbox.core.examples.android.internal.DropboxOAuthUtil
import com.dropbox.core.examples.android.internal.OpenWithActivity
import com.dropbox.core.v2.users.FullAccount

/**
 * Activity that shows information about the currently logged in user
 */
class UserActivity : DropboxActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_user)

        val toolbar = findViewById<Toolbar>(R.id.app_bar)
        setSupportActionBar(toolbar)
        val loginButton = findViewById<Button>(R.id.login_button)
        loginButton.setOnClickListener { v: View? ->
            dropboxOAuthUtil.startDropboxAuthorization(this)
        }
        val filesButton = findViewById<Button>(R.id.files_button)
        filesButton.setOnClickListener { v: View? ->
            startActivity(
                FilesActivity.getIntent(
                    this@UserActivity,
                    ""
                )
            )
        }
        val openWithButton = findViewById<Button>(R.id.open_with)
        openWithButton.setOnClickListener { v: View? ->
            val openWithIntent = Intent(this@UserActivity, OpenWithActivity::class.java)
            startActivity(openWithIntent)
        }

        dropboxOAuthUtil.showWarningDialogIfAppKeyNotSet(this)
    }

    override fun onResume() {
        super.onResume()
        if (hasToken()) {
            findViewById<View>(R.id.login_button).visibility = View.GONE
            findViewById<View>(R.id.email_text).visibility = View.VISIBLE
            findViewById<View>(R.id.name_text).visibility = View.VISIBLE
            findViewById<View>(R.id.type_text).visibility = View.VISIBLE
            findViewById<View>(R.id.files_button).isEnabled = true
            findViewById<View>(R.id.open_with).isEnabled = true
        } else {
            findViewById<View>(R.id.login_button).visibility = View.VISIBLE
            findViewById<View>(R.id.email_text).visibility = View.GONE
            findViewById<View>(R.id.name_text).visibility = View.GONE
            findViewById<View>(R.id.type_text).visibility = View.GONE
            findViewById<View>(R.id.files_button).isEnabled = false
            findViewById<View>(R.id.open_with).isEnabled = false
        }
    }

    override fun loadData() {
        GetCurrentAccountTask(
            dropboxApi.dropboxClient,
            object : GetCurrentAccountTask.Callback {
                override fun onComplete(result: FullAccount?) {
                    requireNotNull(result)
                    (findViewById<View>(R.id.email_text) as TextView).text = result.email
                    (findViewById<View>(R.id.name_text) as TextView).text = result.name.displayName
                    (findViewById<View>(R.id.type_text) as TextView).text = result.accountType.name
                }

                override fun onError(e: Exception?) {
                    Log.e(javaClass.name, "Failed to get account details.", e)
                }
            }).execute()
    }
}